File: frontend/src/lib/classification.ts
import { Verdict } from '../types';

export const classifyResult = (aiProb: number, manipProb: number): Verdict => {
  if (aiProb > 70) return 'AI_GENERATED';
  if (manipProb > 60) return 'MANIPULATED';
  if (aiProb < 30 && manipProb < 30) return 'LIKELY_AUTHENTIC';
  return 'UNCERTAIN';
};

export const getVerdictColor = (verdict: Verdict) => {
  switch (verdict) {
    case 'AI_GENERATED':
      return 'bg-beige-200/10 text-beige-200 border-beige-200/30';
    case 'LIKELY_AUTHENTIC':
      return 'bg-beige-100/10 text-beige-100 border-beige-100/25';
    case 'MANIPULATED':
      return 'bg-beige-400/10 text-beige-300 border-beige-400/30';
    case 'UNCERTAIN':
      return 'bg-beige-500/10 text-beige-500 border-beige-500/25';
    default:
      return 'bg-dark-800 text-beige-400 border-beige-500/20';
  }
};

export const getVerdictLabel = (verdict: Verdict) => {
  switch (verdict) {
    case 'AI_GENERATED':
      return 'AI Generated';
    case 'LIKELY_AUTHENTIC':
      return 'Likely Authentic';
    case 'MANIPULATED':
      return 'Manipulated';
    case 'UNCERTAIN':
      return 'Uncertain';
    default:
      return 'Unknown';
  }
};

export const getConfidenceLabel = (confidence: string) => {
  return confidence.charAt(0).toUpperCase() + confidence.slice(1);
};

File: backend/app/utils/classification.py
def classify_result(ai_probability: int, manipulation_probability: int) -> tuple[str, str]:
    """Returns (verdict, confidence)"""
    if manipulation_probability >= 70:
        verdict = "MANIPULATED"
        confidence = "high" if manipulation_probability >= 85 else "medium"
    elif ai_probability >= 90:
        verdict = "AI_GENERATED"
        confidence = "high"
    elif ai_probability >= 70:
        verdict = "AI_GENERATED"
        confidence = "medium" if ai_probability >= 80 else "low"
    elif ai_probability <= 30:
        verdict = "LIKELY_AUTHENTIC"
        confidence = "high" if ai_probability <= 15 else "medium"
    else:
        verdict = "UNCERTAIN"
        confidence = "low"
    return verdict, confidence
